store/mongo/datastore_mongo.go

```diff
@@ -560,7 +560,7 @@ func (db *DataStoreMongo) getReleases_1_2_14(
 			{Key: "$match", Value: bson.M{
 				StorageKeyImageName: bson.M{
 					"$regex": primitive.Regex{
-						Pattern: ".*" + regexp.QuoteMeta(filt.Name) + ".*",
+						Pattern: regexp.QuoteMeta(filt.Name),
```

Comment:
In the UI it says "filter by prefix". Shouldn't it rather be like this?
```suggestion
						Pattern: "^" + regexp.QuoteMeta(filt.Name),
```

Reply:
I know, and I do not have a problem with changing it, although it will break the undocumented and probably unused API. @kjaskiewiczz opinions?

Reply:
IMO it is fixing a bug. We're saying it is prefix search, but it is returning documents that simply contain the query as a substring. This is leading to confusing/inconsistent results.

Reply:
the UI says "starts with", the API was wrongly documented, so I guess we can fix it by searching for prefix

Reply:
the task which this is a spinoff from was about clarifying this behaviour, and thats what I did, described how it works now. I can fix this PR and the docs PR, no problem.

Reply:
okey then. let me align both of the PRs.

Reply:
I agree, I would fix it making it work as a prefix search.

Reply:
okey so as we all agree. I adjusted:
1. name is searched always by prefix
2. description and type searched anywhere (I honestly do not see UI using it at all)